backend/src/core/services/s3-client.class.ts

```diff
@@ -30,13 +31,31 @@ export default class S3Client {
   /**
    * Download CSV file from S3 and process it into message.
    * The messages are formed from the template and parameters specified in the csv.
-   *
+   * Retries when s3 can't find the s3Key. This could be due to s3's eventual consistency.
    * @param campaignId
    * @param s3Key
    */
   async getCsvFile(s3Key: string): Promise<Array<CSVParams>> {
-    const downloadStream = this.download(s3Key)
-    const fileContents = await ParseCsvService.parseCsv(downloadStream)
-    return fileContents
+    return await retry(
+      async (bail) => {
+        try {
+          const downloadStream = this.download(s3Key)
+          const fileContents = await ParseCsvService.parseCsv(downloadStream)
+          return fileContents
+        } catch (e) {
+          if (e.code !== 'NoSuchKey') {
+            bail(e)
+            return []
```

Comment:
Not sure if this is a good pattern to be using. `retry` will eventually throw an error, even though i am returning an empty array. Initially i wanted to just return void, but this would mean i have to change the signature of the function.

Reply:
This is ok. If it can't get the csv file after retrying, then it should fail and require the user to re-upload.